File: Data Structures & Algorithms/validate-parentheses/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        const stack = [];
        if (s.length < 2) return false;

        for (let i = 0; i < s.length; i++) {
            if (s[i] === "(" || s[i] === "[" || s[i] === "{") {
                stack.push(s[i]);
            } else {
                if (s[i] === ")" && stack.pop() !== "(") return false;
                if (s[i] === "]" && stack.pop() !== "[") return false;
                if (s[i] === "}" && stack.pop() !== "{") return false;
            }
        }
        if (stack.length !== 0) return false;
        return true;
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = [0] * len(temperatures)
        stack = []
        for idx, val in enumerate(temperatures):
            if len(stack) == 0:
                stack.append(idx)
            while stack and temperatures[stack[-1]] < val:
                temp_idx = stack.pop()
                res[temp_idx] = idx - temp_idx
            stack.append(idx)
        return res
